### PR TITLE
Add Lightbox

### PR DESCRIPTION
**Pull request by AdamDotCom**
### Moved the Lightbox from Zero Web into the Component Library.

Lightbox in Zero Web: https://github.com/m3m3n70/zero-web/blob/development/src/components/lightbox/index.tsx
Lightbox tests in Zero Web: https://github.com/m3m3n70/zero-web/blob/development/src/test/component/unit/components/lightbox/lightbox.tsx

Lightbox-Template-⋅-Storybook

### CODE REVIEW

File: src/components/lightbox/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    wrapper.find(ReactImageLightbox).simulate('movePrevRequest');
+    expect(wrapper.find(ReactImageLightbox).prop('mainSrc')).toEqual(cloudinaryify(items[2].url));
+
+    wrapper.find(ReactImageLightbox).simulate('movePrevRequest');
+    expect(wrapper.find(ReactImageLightbox).prop('mainSrc')).toEqual(cloudinaryify(items[1].url));
+
+    wrapper.find(ReactImageLightbox).simulate('movePrevRequest');
+    expect(wrapper.find(ReactImageLightbox).prop('mainSrc')).toEqual(cloudinaryify(items[0].url));
+  });
```
Comment (jtulloch) [inline src/components/lightbox/index.test.tsx]: this is just a comment. don't bother changing now. it's easy enough to split them up later.
however -- i much prefer a single assertion per test for unit tests. it makes the intent of the test more clear, and makes failures much easier to track down. for these ones, i'd probably split them in to two tests each. one to test that the callback increases/decreases the current image, and one that tests the wrap-around condition.

File: src/components/lightbox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  moveNextLightBox = () => {
+    this.setState({ index: (this.state.index + 1) % this.props.items.length });
+  }
+
+  get items() {
+    const { items, provider } = this.props;
+    return items.map(media => {
+      if (media.type === 'image') {
+        const options = provider.fitWithinBox(media, 2000, 1500);
```
Comment (jtulloch) [inline src/components/lightbox/index.tsx]: where do 2000 and 1500 come from?

File: src/components/lightbox/index.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    this.props.onClose();
+  }
+
+  movePrevLightBox = () => {
+    this.setState({ index: (this.state.index + this.props.items.length - 1) % this.props.items.length });
+  }
+
+  moveNextLightBox = () => {
+    this.setState({ index: (this.state.index + 1) % this.props.items.length });
+  }
```
Comment (jtulloch) [inline src/components/lightbox/index.tsx]: we do this same calculation in two places. is it possible to combine them to reduce the chance of errors? something like this:
```
get previousIndex() {
  return (this.state.index + this.props.items.length - 1) % this.props.items.length;
}

get previousItem() {
  const { items } = this.props;
  if (items.length <= 1) return null;
  
  return items[this.previousIndex];
}

movePrevLightBox = () => {
  this.setState({ index: this.previousIndex });
}

```

**Review by jtulloch — CHANGES_REQUESTED** [review]

File: src/components/lightbox/index.tsx (hunk 2 above)
Comment (AdamDotCom) [reply src/components/lightbox/index.tsx]: Rich would know. https://github.com/m3m3n70/zero-web/blame/development/src/components/lightbox/index.tsx#L52

Comment (AdamDotCom) [reply src/components/lightbox/index.tsx]: I guess we can drop them.

File: src/components/lightbox/index.test.tsx (hunk 1 above)
Comment (AdamDotCom) [reply src/components/lightbox/index.test.tsx]: They've been like that for the [past five years](https://github.com/m3m3n70/zero-web/blame/development/src/test/component/unit/components/lightbox/lightbox.tsx#L60). We'll thank our future selves if we make the change today.

**Pull request merged**